server/metric_server.go

```diff
@@ -0,0 +1,76 @@
+package server
+
+import (
+	"calendar/storage"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"runtime"
+	"strconv"
+)
+
+func NewMetricServer(store storage.EventStore) *EventServer {
+	ms := new(EventServer)
+
+	ms.Store = store
+	ms.UserStore = storage.Users
+
+	router := http.NewServeMux()
+	router.HandleFunc("/metric/events", ms.TotalEvents)
+	router.HandleFunc("/metric/users", ms.TotalUsers)
+	router.HandleFunc("/metric/requests_per_s", ms.Empty)
+	router.HandleFunc("/metric/requests_per_s_per_u", ms.Empty)
+	router.HandleFunc("/metric/goroutines", ms.TotalGoroutines)
+	router.HandleFunc("/metric/memory", ms.TotalMemory)
+	router.HandleFunc("/metric/cpu", ms.TotalCPU)
+
+	ms.Handler = router
+
+	return ms
+}
+func (ms *EventServer) Empty(w http.ResponseWriter, r *http.Request) {
+
+}
+func (ms *EventServer) TotalEvents(w http.ResponseWriter, r *http.Request) {
+	_, err := w.Write([]byte("Number of events: " + strconv.Itoa(ms.Store.Count())))
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+}
+
+func (ms *EventServer) TotalUsers(w http.ResponseWriter, r *http.Request) {
+	_, err := w.Write([]byte("Number of users: " + strconv.Itoa(ms.UserStore.Count())))
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+}
+
+func (ms *EventServer) TotalGoroutines(w http.ResponseWriter, r *http.Request) {
+	_, err := w.Write([]byte("Number of goroutines: " + strconv.Itoa(runtime.NumGoroutine())))
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+}
+
+func (ms *EventServer) TotalMemory(w http.ResponseWriter, r *http.Request) {
+	var m runtime.MemStats
+	runtime.ReadMemStats(&m)
+	fmt.Println(m.Alloc)
+	w.Header().Set("content-type", jsonContentType)
+	err := json.NewEncoder(w).Encode(m)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+}
+
+func (ms *EventServer) TotalCPU(w http.ResponseWriter, r *http.Request) {
+	_, err := w.Write([]byte("Number of logical CPUs usable by the current process: " + strconv.Itoa(runtime.NumCPU()) + "\n" + "number of cgo calls made by the current process: " + strconv.Itoa(int(runtime.NumCgoCall()))))
```

Comment:
This is the ideal place to use text/template library

Reply:
I thought of it and maybe I'll add it for the next pull request